coclico/metrics/metric.py

```diff
@@ -6,19 +6,76 @@
+        These jobs are aimed to compute intermediate results relative to the input las only (without comparison) that
+        will be used in `create_metric_relative_to_ref_jobs` to compute the relative metric
+        Args:
+            name (str): classification name (used for job name creation)
+            tile_names (List[str]): list of the filenames of the tiles on which to calculate the result
+            input_path (Path): input folder path (path to the results of the classification)
+            out_path (Path): path for the intermediate results to be saved
+            is_ref (bool): flag that says if the input classification folder is the reference folder (in case it has to
+            be treated differently)
+
+        Returns:
+            List[Job]: List of GPAO jobs to create
+        """
+        return [self.create_metric_intrinsic_one_job(name, input_path / f, out_path, is_ref) for f in tile_names]
+
+    def create_metric_intrinsic_one_job(self, name: str, input: Path, output: Path, is_ref: bool) -> Job:
+        """Create a job for a single input classification result file (one tile).
+        ie. job to create of one element of the `tiles_names` list in create_metric_intrinsic_jobs
 
-    def create_metric_intrinsic_one_job(self, name: str, input: Path, output: Path) -> Job:
+        Args:
+            name (str): classification name (used for job name creation)
+            input (Path): full path of the input tile
+            output (Path): output folder for the result
+            is_ref (bool):  flag that says if the input classification folder is the reference folder (in case it has
+            to be treated differently)
+
+        Raises:
+            NotImplementedError: should be implemented in children classes
+
+        Returns:
+            Job: GPAO job to create
+        """
         raise NotImplementedError
 
     def create_metric_relative_to_ref_jobs(
         self, name: str, out_c1: Path, out_ref: Path, output: Path, c1_jobs: List[Job], ref_jobs: List[Job]
     ) -> List[Job]:
+        """Create jobs to generate a metric that compares the classification results from a c1 folder and th ref folder
+        based on the results of the results of the jobs created by create_metric_intrinsic_jobs
+        Args:
+            name (str):  classification name (used for job name creation)
+            out_c1 (Path): path to the results of the intrinsic metric jobs for c1 (input for this part)
+            out_ref (Path): path to the results of the intrinsic metric jobs for ref (input for this part)
+            output (Path): path to the results of the relative metric
+            c1_jobs (List[Job]): Jobs created in create_metric_intrinsic_jobs for c1 (on which the jobs created in this
+            method will depend)
+            ref_jobs (List[Job]): Jobs created in create_metric_intrinsic_jobs for ref (on which the jobs created in
```

Comment:
Ce qui est entre parenthese est en fait très important : le développeur qui ajoute une métrique ne doit pas oublier de le faire. J'aurai bien écrit quelque chose de plus alertant, genre une seconde ligne, comme ceci :

`Note: Each new jobs must depends on these ref jobs, to ensure it will be computed after.`

Ce commentaire s'applique aussi au param c1_jobs